src/api/spec/models/package_spec.rb

```diff
@@ -98,4 +102,74 @@
       end
     end
   end
+
+  context '#maintainers' do
+    before do
+      User.current = user
+    end
+
+    it 'returns an array with user objects to all maintainers for a package' do
+      # first of all, we add a user who is not a maintainer but a bugowner
+      # he/she should not be recognized by package.maintainers
+      package.add_user(other_user2, Role.find_by_title!("bugowner"))
+      package.save!
+
+      # we expect to get one user object in an array
+      package.add_user(user, Role.find_by_title!("maintainer"))
+      package.save!
+
+      expect(package.maintainers).to eq([user])
+
+      # we expect the previous user and the following user to be in that array
+      package.add_user(other_user, Role.find_by_title!("maintainer"))
+      package.save!
+
+      expect(package.maintainers).to eq([other_user, user])
+    end
+
+    it 'resolves groups properly' do
+      # groups should be resolved and only their assigned users should be in the
+      # returning array
+      group.add_user(other_user)
+      group.add_user(other_user2)
+
+      # add a group to the package what is not a maintainer to make sure it'll
+      # be ignored when calling package.maintainers
+      group_bugowner.add_user(create(:confirmed_user, login: 'tom5'))
+
+      package.add_group(group_bugowner, Role.find_by_title!("bugowner"))
+      package.save
+
+      package.add_group(group, Role.find_by_title!("maintainer"))
+      package.save
+
+      expect(package.maintainers).to eq([other_user, other_user2])
+    end
+
+    it 'makes sure that no user is listed more than one time' do
+      group.add_user(user)
+      group_bugowner.add_user(user)
+
+      package.add_group(group, Role.find_by_title!("maintainer"))
+      package.add_group(group_bugowner, Role.find_by_title!("maintainer"))
+      package.add_user(user, Role.find_by_title!("maintainer"))
+
+      package.save
+
+      expect(package.maintainers).to eq([user])
+    end
+
+    it 'returns users and the users of resolved groups' do
+      group.add_user(user)
+      group_bugowner.add_user(other_user)
+
+      package.add_group(group, Role.find_by_title!("maintainer"))
+      package.add_group(group_bugowner, Role.find_by_title!("maintainer"))
+      package.add_user(other_user2, Role.find_by_title!("maintainer"))
+
+      package.save
+
+      expect(package.maintainers).to contain_exactly(user, other_user, other_user2)
```

Comment:
`contain_exactly` is the same as `match_array` in the end.... so we should use only one :smile: